src/server/game/Entities/Unit/Unit.cpp

```diff
@@ -12605,8 +12605,39 @@ uint32 Unit::GetModelForForm(ShapeshiftForm form) const
     return modelid;
 }
 
-uint32 Unit::GetModelForTotem(PlayerTotemType totemType)
+uint32 Unit::GetCategoryForTotem(uint32 totemCategory)
+{
+    switch (totemCategory)
+    {
+        case TOTEM_CATEGORY_FIRE:
+        case TOTEM_CATEGORY_FIRE_2:
+        case TOTEM_CATEGORY_FIRE_3:
+            return SUMMON_TYPE_TOTEM_FIRE;
+        case TOTEM_CATEGORY_EARTH:
+        case TOTEM_CATEGORY_EARTH_2:
+        case TOTEM_CATEGORY_EARTH_3:
+        case TOTEM_CATEGORY_EARTH_4:
+            return SUMMON_TYPE_TOTEM_EARTH;
+        case TOTEM_CATEGORY_WATER:
+        case TOTEM_CATEGORY_WATER_2:
+            return SUMMON_TYPE_TOTEM_WATER;
+        case TOTEM_CATEGORY_AIR:
+        case TOTEM_CATEGORY_AIR_2:
+        case TOTEM_CATEGORY_AIR_3:
+        case TOTEM_CATEGORY_AIR_4:
+        case TOTEM_CATEGORY_AIR_5:
+        case TOTEM_CATEGORY_AIR_6:
+            return SUMMON_TYPE_TOTEM_AIR;
+        default:
+            break;
+    }
+    return 0;
```

Comment:
that would mean, all other totemcategories are firetotems

Reply:
Fixed with last commit, i don't have see but TotemCategory are already named in SharedDefine. I know this PR are not 100% blizz because i have two options, create hotfix in SpellTotems.db2 for add category of totem for spell player and this method. Also normaly displayid are done by a serverside spell. So much data in hotfix table...

Reply:
Fixed with last commit, i don't have see but TotemCategory are already named in SharedDefine. I know this PR are not 100% blizz because i have two options, create hotfix in SpellTotems.db2 for add category of totem for spell player and this method. Also normaly displayid are done by a serverside spell. So much data in hotfix table...